### PR TITLE
fix(#patch); aurigami; add rewards

### PR DESCRIPTION
**Pull request by dmelotik**
testing: https://okgraph.xyz/?q=dmelotik%2Faurigami-aurora

#1273

### CODE REVIEW

File: subgraphs/compound-forks/protocols/aurigami/src/mapping.ts [hunk 1]
```diff
@@ -333,6 +350,184 @@ function getOrCreateProtocol(): LendingProtocol {
...
+    ]);
+    return;
+  }
+
+  const rewardTokens: string[] = [];
+  const rewardsAmount: BigInt[] = [];
+  const rewardsAmountUSD: BigDecimal[] = [];
+
+  // PLY Borrow first since Reward tokens will be alphabetized
+  // Ply Borrow, Ply Supply, AURORA Borrow, AURORA Supply
```
Comment (jaimehgb) [inline subgraphs/compound-forks/protocols/aurigami/src/mapping.ts:379]: Shouldn’t Aurora go before PLY then?

File: subgraphs/compound-forks/protocols/aurigami/src/mapping.ts [hunk 2]
```diff
@@ -333,6 +350,184 @@ function getOrCreateProtocol(): LendingProtocol {
...
+    const priceUSD = getPrice(AURORA_MARKET, protocol._priceOracle)
+      .value.toBigDecimal()
+      .div(mantissaFactorBD);
+    rewardsAmount.push(rewards.rewardsPerDayBI);
+    rewardsAmountUSD.push(rewards.rewardsPerDayBD.times(priceUSD));
+    log.warning("price: {}, token: {}", [
+      priceUSD.toString(),
+      rewards.token.name,
+    ]);
+  }
```
Comment (jaimehgb) [inline subgraphs/compound-forks/protocols/aurigami/src/mapping.ts:476]: If you want to reduce duplication you can do something like:

```
function calculateMarketRewards(token: Address, speed: BigInt, oracle: Address) {
  if (!speed.gt(BIGINT_ONE)) { return; }
  // get or create reward token
  // get rewards per day
  // get price use
  // rewardTokens.push(), rewardAmount.push() ….
  // etc
}
```

File: subgraphs/compound-forks/protocols/aurigami/src/mapping.ts (hunk 1 above)
Comment (dmelotik) [reply subgraphs/compound-forks/protocols/aurigami/src/mapping.ts:379]: based off of the token address not the name

File: subgraphs/compound-forks/protocols/aurigami/src/mapping.ts (hunk 2 above)
Comment (dmelotik) [reply subgraphs/compound-forks/protocols/aurigami/src/mapping.ts:476]: Would I create a class to pass the arrays in and out of the function? THe assemblyscript kinda makes this tricky

File: subgraphs/compound-forks/protocols/aurigami/src/mapping.ts (hunk 1 above)
Comment (jaimehgb) [reply subgraphs/compound-forks/protocols/aurigami/src/mapping.ts:379]: Hmm, but the `rewardToken.id` includes the reward type. So then it should be PLY Borrow, Aurora Borrow, PLY Supply, Aurora Supply?

File: subgraphs/compound-forks/protocols/aurigami/src/mapping.ts (hunk 2 above)
Comment (dmelotik) [reply subgraphs/compound-forks/protocols/aurigami/src/mapping.ts]: done

**Review by jaimehgb — APPROVED** [review]

**Comment by ishraq8** [comment]
Double checked reward, they are close to what is on their mainsite

**Pull request merged**